src/main/kotlin/dev/minn/jda/ktx/InlineShardManagerBuilder.kt

```diff
@@ -0,0 +1,542 @@
+    var context: Boolean = true
+        set(value) {
+            builder.setContextEnabled(value)
+            field = value
+        }
+    
+    var compression: Compression? = null
+        set(value) {
+            if (value != null) {
+                builder.setCompression(value)
+            }
+            field = value
+        }
+    
+    val eventListeners: MutableCollection<Any> = DelegatingCollection(
+            adder = { item -> builder.addEventListeners(item) },
+            remover = { item -> builder.removeEventListeners(item) },
+                                                                     )
+    
+    val eventListenerProvider: MutableCollection<IntFunction<Any>> = DelegatingCollection(
+            adder = { item -> builder.addEventListenerProvider(item) },
+            remover = { item -> builder.removeEventListenerProvider(item) },
+                                                                                         )
+    
+    var audioSendFactory: IAudioSendFactory? = null
+        set(value) {
+            builder.setAudioSendFactory(value)
+            field = value
+        }
+    
+    var autoReconnect: Boolean = true
+        set(value) {
+            builder.setAutoReconnect(value)
+            field = value
+        }
+    
+    var bulkDeleteSplitting: Boolean = true
+        set(value) {
+            builder.setBulkDeleteSplittingEnabled(value)
+            field = value
+        }
+    
+    var enableShutdownHook: Boolean = true
+        set(value) {
+            builder.setEnableShutdownHook(value)
+            field = value
+        }
+    
+    @Deprecated("Use eventManagerProvider instead")
+    var eventManager: IEventManager? = null
+        set(value) {
+            if (value != null) {
+                eventManagerProvider = IntFunction { value }
+            }
+            field = value
+        }
+    
+    var eventManagerProvider: IntFunction<out IEventManager>? = null
+        set(value) {
+            if (value != null) {
+                builder.setEventManagerProvider(value)
+            }
+            field = value
+        }
+    
+    var activity: Activity? = null
+        set(value) {
+            builder.setActivity(value)
+            field = value
+        }
+    
+    var activityProvider: IntFunction<Activity>? = null
+        set(value) {
+            builder.setActivityProvider(value)
+            field = value
+        }
+    
+    var idle: Boolean = false
+        set(value) {
+            builder.setIdle(value)
+            field = value
+        }
+    
+    var idleProvider: IntFunction<Boolean>? = null
+        set(value) {
+            builder.setIdleProvider(value)
+            field = value
+        }
+    
+    var status: OnlineStatus? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatus(value)
+            }
+            field = value
+        }
+    
+    var statusProvider: IntFunction<OnlineStatus>? = null
```

Comment:
I think it would be better to have a pattern like this:

```kotlin
status(ONLINE)

status { 
  if (it == 1) ONLINE
  else OFFLINE
}
```

instead of the current property syntax which assigns functions you have here.

Reply:
Hmmm. The second one looks nice.
What about the following?
```kotlin
status = ONLINE

status {
    if (it == 1)
        ONLINE
    else
        OFFLINE
}
```